tests/utils/scribereader_test.py

```diff
@@ -0,0 +1,144 @@
+import datetime
+from unittest import mock
+
+import pytest
+
+from tron.utils.scribereader import read_log_stream_for_action_run
+
+try:
+    import scribereader  # noqa: F401
+except ImportError:
+    pytest.skip("scribereader not available, skipping tests", allow_module_level=True)
+
+
+def test_read_log_stream_for_action_run_min_date_and_max_date_today():
+    with mock.patch(
+        "tron.utils.scribereader.get_scribereader_host_and_port", autospec=True, return_value=("host", 1234),
+    ), mock.patch(
+        "tron.utils.scribereader.scribereader.get_stream_reader", autospec=True,
+    ) as mock_stream_reader, mock.patch(
+        "tron.utils.scribereader.scribereader.get_stream_tailer", autospec=True,
+    ) as mock_stream_tailer:
+        # in this case, we shouldn't even try to check the reader, so lets set an exception
+        # to make sure we didn't try
+        mock_stream_reader.return_value.__enter__.side_effect = Exception
+        # but the tailer should have some data and actually be used
+        mock_stream_tailer.return_value.__iter__.return_value = iter(
+            [
+                """{
+                "tron_run_number": 1234,
+                "component": "stdout",
+                "message": "line 1",
+                "timestamp": "2021-01-02T18:10:09.169421619Z"
+            }""",
+                """{
+                "tron_run_number": 1234,
+                "component": "stdout",
+                "message": "line 2",
+                "timestamp": "2021-01-02T18:11:09.169421619Z"
+            }""",
+                """{
+                "tron_run_number": 1234,
+                "component": "stderr",
+                "message": "line 3",
+                "timestamp": "2021-01-02T18:12:09.169421619Z"
+            }""",
+            ]
+        )
+        output = read_log_stream_for_action_run(
+            action_run_id="namespace.job.1234.action",
+            component="stdout",
+            min_date=datetime.datetime.now(),
```

Comment:
nit: calling `now()` looks sketch because it makes the tests look like they depend on the current time, even though its not the case because they are just forwarded to scribereader. id either use as specific datetime or (more likely) comment explaining why this is fine.

Reply:
true! I'll add a comment